File: go/tactical-agent/agent_card.go
package main

// agent_card.go — the static Agent Card for the Tactical Analysis Agent (§3,
// §12.4). The structure mirrors the other Go agents; only the name, description,
// url, capabilities, and skills differ.

const (
	agentName   = "tactical-agent"
	defaultPort = "8021"
	agentURL    = "http://tactical-agent:8021"
)

func buildAgentCard() AgentCard {
	return AgentCard{
		Name:        agentName,
		Description: "Resistance tactical analysis agent — deterministic risk scoring and strategy.",
		Provider: AgentProvider{
			Organization: "Rebel Alliance",
			Url:          "https://resistance.local",
		},
		Version: "1.0.0",
		Url:     agentURL,
		Capabilities: AgentCapabilities{
			Streaming:         false,
			PushNotifications: false,
		},
		DefaultInputModes:  []string{"text/plain", "application/json"},
		DefaultOutputModes: []string{"text/plain", "application/json"},
		Skills: []AgentSkill{
			{
				Id:          "calculate_risk",
				Name:        "Calculate Risk",
				Description: "Computes a deterministic threat/risk score from an intelligence report.",
				InputModes:  []string{"text/plain", "application/json"},
				OutputModes: []string{"application/json"},
				Examples:    []string{"Calculate the risk of the Imperial buildup at Hoth"},
			},
			{
				Id:          "generate_strategy",
				Name:        "Generate Strategy",
				Description: "Produces a recommended defensive strategy for the engagement.",
				InputModes:  []string{"text/plain", "application/json"},
				OutputModes: []string{"application/json"},
				Examples:    []string{"Generate a defensive strategy for Echo Base"},
			},
			{
				Id:          "prioritize_targets",
				Name:        "Prioritize Targets",
				Description: "Ranks priority targets by tactical value.",
				InputModes:  []string{"text/plain", "application/json"},
				OutputModes: []string{"application/json"},
				Examples:    []string{"Prioritize the Imperial targets at Hoth"},
			},
			{
				Id:          "recommend_action",
				Name:        "Recommend Action",
				Description: "Recommends a course of action given the assessed threat level.",
				InputModes:  []string{"text/plain", "application/json"},
				OutputModes: []string{"application/json"},
				Examples:    []string{"Recommend an action for the HIGH threat at Hoth"},
			},
		},
		SecuritySchemes: map[string]SecurityScheme{
			"demoApiKey": {Type: "apiKey", In: "header", Name: "X-Demo-Token"},
		},
		Security: []map[string][]string{
			{"demoApiKey": {}},
		},
	}
}

File: typescript/intelligence-agent/src/agentCard.ts
// Agent Card for the Intelligence Agent (§3 / §12.2).
// Returned verbatim at GET /.well-known/agent-card.json (no auth).

export const AGENT_NAME = "intelligence-agent";
export const AGENT_LANGUAGE = "typescript";

/** The agent's own base URL (docker-compose service name + port). */
const SELF_URL = `http://${AGENT_NAME}:8011`;

export const agentCard = {
  name: AGENT_NAME,
  description: "Resistance intelligence scouting agent.",
  provider: { organization: "Rebel Alliance", url: "https://resistance.local" },
  version: "1.0.0",
  url: SELF_URL,
  capabilities: { streaming: true, pushNotifications: false },
  defaultInputModes: ["text/plain", "application/json"],
  defaultOutputModes: ["text/plain", "application/json"],
  skills: [
    {
      id: "scout_system",
      name: "Scout Star System",
      description: "Scans a star system for Imperial activity.",
      inputModes: ["text/plain", "application/json"],
      outputModes: ["application/json"],
      examples: ["Scout Hoth for Imperial movement"],
    },
    {
      id: "detect_empire_presence",
      name: "Detect Empire Presence",
      description: "Detects whether Imperial forces are present in a system.",
      inputModes: ["text/plain", "application/json"],
      outputModes: ["application/json"],
      examples: ["Detect Empire presence near Hoth"],
    },
    {
      id: "estimate_force_strength",
      name: "Estimate Force Strength",
      description: "Estimates the size and composition of detected Imperial forces.",
      inputModes: ["text/plain", "application/json"],
      outputModes: ["application/json"],
      examples: ["Estimate Imperial force strength on Hoth"],
    },
    {
      id: "produce_intelligence_report",
      name: "Produce Intelligence Report",
      description: "Compiles a full intelligence report for a scouted system.",
      inputModes: ["text/plain", "application/json"],
      outputModes: ["application/json"],
      examples: ["Produce an intelligence report for the Hoth system"],
    },
  ],
  securitySchemes: {
    demoApiKey: { type: "apiKey", in: "header", name: "X-Demo-Token" },
  },
  security: [{ demoApiKey: [] }],
} as const;

/** Skill ids advertised by this agent (used in the startup banner). */
export const SKILL_IDS = agentCard.skills.map((s) => s.id);
